File: src/app/models/producto.py
from sqlalchemy import Boolean, Column, DateTime, Float, Integer, String, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.db import Base


class Producto(Base):
    """Producto: comida de animales vendida por kg."""

    __tablename__ = "productos"

    id = Column(Integer, primary_key=True, index=True)
    nombre = Column(String(255), nullable=False)
    descripcion = Column(Text, nullable=True)
    precio_por_kg = Column(Float, nullable=False)
    categoria = Column(String(100), nullable=True)  # perro, gato, snack
    imagen_url = Column(String(500), nullable=True)
    activo = Column(Boolean, default=True, nullable=False)
    creado_en = Column(DateTime(timezone=True), server_default=func.now())
    actualizado_en = Column(DateTime(timezone=True), onupdate=func.now())

    carrito = relationship("Carrito", back_populates="producto")
    depositos = relationship("Deposito", back_populates="producto")
    pedido_detalles = relationship("PedidoDetalle", back_populates="producto")
    comentarios = relationship("Comentario", back_populates="producto")

File: client/src/types/producto.ts
/**
 * Tipos para productos (comida de animales por kg). Alineados al backend.
 */
export interface Producto {
  id: number;
  nombre: string;
  precio_por_kg?: number;
  descripcion?: string;
  categoria_producto?: {
    id: number;
    nombre: string;
  };
  imagen_url?: string;
  activo?: boolean;
}

export interface IProductoSimple {
  id: number;
  nombre: string;
}

export interface ProductoCreate {
  nombre: string;
  precio_por_kg: number;
  descripcion?: string;
  /** ID de categoría (frontend) o objeto completo para la API */
  categoria_producto?: { nombre: string; descripcion?: string };
  imagen_url?: string;
  activo?: boolean;
}

export interface ProductoPatch {
  nombre?: string;
  precio_por_kg?: number;
  descripcion?: string;
  categoria_producto?: number | { id: number; nombre: string; descripcion?: string };
  imagen_url?: string;
  activo?: boolean;
}

export interface ProductoUpdate {
  nombre?: string;
  precio_por_kg?: number;
  descripcion?: string;
  categoria_producto?: number;
  imagen_url?: string;
  activo?: boolean;
}